packages/instant-meilisearch/src/adapter/search-request-adapter/filter-adapter.ts

```diff
@@ -1,93 +1,155 @@
+ * Escape backslash (\\) and quote (") characters.
+ *
+ * @param {string} filter
+ * @returns {string}
  */
-function transformFilter(filter: string): string {
-  return filter.replace(/(.*):(.*)/i, '"$1"="$2"')
+function transformNumericFilter(filter: string): string {
+  // TODO: What if escape facet values is enabled?
+  //       https://github.com/algolia/instantsearch/blob/da701529ed325bb7a1d782e80cb994711e20d94a/packages/instantsearch.js/src/lib/utils/escapeFacetValue.ts#L13-L21
+  const [attribute, operator, value] = filter.split(numericSplitRegExp)
+  const escapedAttribute = getValueWithEscapedBackslashesAndQuotes(attribute)
+  return `"${escapedAttribute.trim()}"${
+    operator === ':' ? ' ' : operator
+  }${value.trim()}`
 }
 
 /**
- * Itterate over all filters.
+ * Iterate over all filters.
  * Return the filters in a Meilisearch compatible format.
+ * Can return empty string or empty array, but not an array of empty strings
+ * or empty arrays.
  *
- * @param  {SearchContext['facetFilters']} filters?
+ * @param  {(filter: string) => string} transformCallback
+ * @param  {SearchContext['facetFilters']} filters
  * @returns {Filter}
  */
-function transformFilters(filters?: SearchContext['facetFilters']): Filter {
-  if (typeof filters === 'string') {
-    return transformFilter(filters)
-  } else if (Array.isArray(filters))
-    return filters
-      .map((filter) => {
-        if (Array.isArray(filter))
-          return filter
-            .map((nestedFilter) => transformFilter(nestedFilter))
-            .filter((elem) => elem)
-        else {
-          return transformFilter(filter)
-        }
-      })
-      .filter((elem) => elem)
-  return []
+function transformFilters(
+  transformCallback: (filter: string) => string,
+  filters: NonNullable<SearchContext['facetFilters']>
+): Filter {
+  // Note on `Array.isArray`:
+  // https://github.com/microsoft/TypeScript/issues/17002
+  // Typescript has problems with readonly, which makes it undesirable in
+  // most situations, the crux of the issue being that `Array.isArray`
+  // will return true for mutable arrays as well as immutable ones. Since Instantsearch.js
+  // decided to use readonly, we have to type cast, while keeping as much of the type
+  // safety as possible.
+  return typeof filters === 'string'
+    ? transformCallback(filters)
+    : filters
+        .map((filter) =>
+          (<(value: readonly any[] | any) => value is readonly any[]>(
+            Array.isArray
+          ))(filter)
+            ? filter
+                .map((nestedFilter) => transformCallback(nestedFilter))
+                // TODO: Do these filters have any purpose? Can we actually get empty strings? Should we handle empty strings?
+                //       Maybe handling these should be the responsibility of instantsearch.js and/or the user and their input.
+                //       Instead of these empty filters quietly being swallowed, they might even get us an error message, and
```

Comment:
Can you provide a use test case for that? Because I agree with you, we should not handle that if instantsearch already do it.

Reply:
What I'm referring to with this, is that under normal circumstances Instantsearch.js shouldn't pass on an empty filter, and if it does, something is clearly wrong. So I propose that we do not check for empty strings, as that might just worsen the issue, because this way we swallow potential errors.

I have to check if by any chance IS.js can return an empty string on purpose, but so far it doesn't seem the case. I'm unsure why the previous maintainer filtered out empty strings, perhaps we can ask him to save us the trouble?